python/pyarrow/src/arrow/python/datetime.cc

```diff
@@ -633,6 +637,7 @@ Result<PyObject*> MonthDayNanoIntervalArrayToPyList(
   OwnedRef out_list(PyList_New(array.length()));
   RETURN_IF_PYERROR();
   PyListAssigner out_objects(out_list.obj());
+  RETURN_IF_PYERROR();
```

Comment:
The `PyListAssigner` does not set any Python exception, so this doesn't seem useful. On the other hand, I assume an exception might be set during `WriteArrayObjects` below, is that checked for?